src/format/OpData01.cpp

```diff
@@ -0,0 +1,143 @@
+            qlen = (ch << (8 * i)) | qlen;
+        }
+
+        // stop kidding ourselves and having to downcast len
+        len = int(qlen);
+    }
+    // qDebug("toLongLong := %lld\n", qlen);
+
+    QByteArray iv(16, '\0');
+    for (int i = 0, iv_len = 16; i < iv_len and it != it_end; ++i, ++it) {
+        char ch = *it;
+        iv[i] = ch;
+    }
+    if (16 != iv.size()) {
+        RETURN_OPDATA01_ERROR_INSTANCE(QString("short IV read, wanted %1 but got %2").arg(16).arg(iv.size()));
+    }
+
+    SymmetricCipher cipher(SymmetricCipher::Aes256, SymmetricCipher::Cbc, SymmetricCipher::Decrypt);
+    if (!cipher.init(key, iv)) {
+        RETURN_OPDATA01_ERROR_INSTANCE(QString("Unable to init cipher for opdata01: %1").arg(cipher.errorString()));
+    }
+
+    QByteArray* clearText;
+    {
+        const int blockSize = cipher.blockSize();
+        // qDebug("cipher.blockSize = %d\n", blockSize);
+
+        int randomBytes = blockSize - (len % blockSize);
+        // qDebug("cipher.randomBytes1 = %d\n", randomBytes);
+        if (0 == randomBytes) {
+            // add random blk
+            randomBytes = blockSize;
+        }
+        int clear_len = len + randomBytes;
+        // qDebug("cipher.randomBytes2 = %d\tclearText.len=%d\n", randomBytes, clear_len);
+        QByteArray qbaCT(clear_len, '\0');
+        for (int i = 0; i < clear_len and it != it_end; ++i, ++it) {
+            qbaCT[i] = *it;
+        }
+
+        const int hmacLen = 256 / 8;
+        QByteArray hmacSig; // 256 / 8, '\0');
+        for (int i = 0; i < hmacLen and it != it_end; ++i, ++it) {
+            hmacSig.append(*it);
+        }
+        if (hmacSig.size() != hmacLen) {
+            RETURN_OPDATA01_ERROR_INSTANCE(
+                QString("Short HMAC sig read; wanted %1 but got %2").arg(hmacLen).arg(hmacSig.size()));
+        }
+
+        if (it < it_end) {
+            qWarning("Did not consume all of opdata01!");
+        } else if (it > it_end) {
+            qWarning("opdata01 iterator greater than const_end");
+        }
+
+        const QByteArray actualHmac =
+            CryptoHash::hmac(data.mid(0, data.size() - hmacSig.size()), hmacKey, CryptoHash::Algorithm::Sha256);
+        if (actualHmac != hmacSig) {
+            RETURN_OPDATA01_ERROR_INSTANCE(QString("opdata01 failed its HMAC; wanted %1 but got %2")
+                                               .arg(QString::fromUtf8(hmacSig.toHex(':')))
+                                               .arg(QString::fromUtf8(actualHmac.toHex(':'))));
+        }
+
+        if (!cipher.processInPlace(qbaCT)) {
+            RETURN_OPDATA01_ERROR_INSTANCE(QString("Unable to process clearText in place"));
+        }
+        const QByteArray& actualClearText = qbaCT.mid(randomBytes);
+        if (actualClearText.size() != len) {
+            RETURN_OPDATA01_ERROR_INSTANCE(
+                QString("Expected %1 bytes of clear-text, found %2").arg(len).arg(actualClearText.size()));
```

Comment:
It's usually called plaintext when it's input to a cipher or output of a decipher.